From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SuperKali <hello@superkali.me>
Date: Wed, 3 Jun 2026 09:38:29 +0800
Subject: arm64: dts: qcom: sc8280xp: Fix GPI DMA0 channel count

The GPI DMA0 controller on SC8280XP only supports 12 GPII instances, not
13. Describing the non-existent 13th GPII makes the driver initialize an
invalid channel and IRQ, which may cause the device to crash and reset
when that GPII is used.

This matches the TrustZone configuration and downstream sources.

Fixes: 71b12166a2be ("arm64: dts: qcom: sc8280xp: Describe GPI DMA controller nodes")
Link: https://git.codelinaro.org/clo/le-au-os/kernel/devicetree/-/blob/LE.UM.7.2.1.c5-00400-sa415m/qcom/direwolf-qupv3.dtsi
Signed-off-by: Xilin Wu <sophon@radxa.com>

(cherry picked from commit c32a46f8766ca2827f4eeca1fe6743c87af8870d)
[SuperKali: backported to sc8280xp edge from radxa/kernel linux-7.0.11]
Signed-off-by: SuperKali <hello@superkali.me>
---
 arch/arm64/boot/dts/qcom/sc8280xp.dtsi | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp.dtsi b/arch/arm64/boot/dts/qcom/sc8280xp.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp.dtsi
+++ b/arch/arm64/boot/dts/qcom/sc8280xp.dtsi
@@ -1359,11 +1359,10 @@ gpi_dma0: dma-controller@900000 {
 				     <GIC_SPI 252 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 253 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 254 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 255 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 256 IRQ_TYPE_LEVEL_HIGH>;
+				     <GIC_SPI 255 IRQ_TYPE_LEVEL_HIGH>;
 
-			dma-channels = <13>;
-			dma-channel-mask = <0x1fff>;
+			dma-channels = <12>;
+			dma-channel-mask = <0xfff>;
 			#dma-cells = <3>;
 
 			iommus = <&apps_smmu 0x576 0x0>;
-- 
Armbian

